### PR TITLE
[BE] Task #208199 Registered in camp feature

### PR DESCRIPTION
**Pull request by GitVivekHub**
https://tracker.tekdi.net/issues/208199

# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

### CODE REVIEW

File: src/src/beneficiaries/beneficiaries.core.service.ts [hunk 1]
```diff
@@ -256,4 +256,48 @@ export class BeneficiariesCoreService {
...
+		);
+	}
+
+	public async getBeneficiaryDetailsById(id, status) {
+		let filter_query = [];
+
+		filter_query.push(`id:{_eq:${id}}`);
+		if (status) {
+			filter_query.push(
+				`program_beneficiaries: {status: {_eq: ${status}}}`,
```
Comment (sagarkoshti1990) [inline src/src/beneficiaries/beneficiaries.core.service.ts:279]: Add cohort specific filter.

File: src/src/camp/camp.service.ts [hunk 2]
```diff
@@ -984,6 +986,28 @@ export class CampService {
...
+					for (const learnerId of learner_ids) {
+						let result =
+							await this.beneficiariesCoreService.getBeneficiaryDetailsById(
+								learnerId,
+								status,
+							);
+						update_beneficiaries_array.push(result);
+					}
+
+					console.log(update_beneficiaries_array);
```
Comment (sagarkoshti1990) [inline src/src/camp/camp.service.ts:1001]: remove console log

**Review by sagarkoshti1990 — CHANGES_REQUESTED** [review]

File: src/src/beneficiaries/beneficiaries.core.service.ts (hunk 1 above)
Comment (GitVivekHub) [reply src/src/beneficiaries/beneficiaries.core.service.ts]: resolved

File: src/src/camp/camp.service.ts (hunk 2 above)
Comment (GitVivekHub) [reply src/src/camp/camp.service.ts]: resolved

**Review by sagarkoshti1990 — APPROVED** [review]

**Pull request merged**